core/controllers/feedback_test.py

```diff
@@ -69,7 +69,7 @@ def test_invalid_exploration_ids_return_page_not_found(self):
 
     def test_invalid_thread_ids_return_empty_message_list(self):
         response_dict = self.get_json(
-            '%s/%s/bad_thread_id' % (
+            '%s/%s.invalid_thread' % (
```

Comment:
Why not %s/invalid_thread_id? Should be stripping the references to exp_id.

(And make sure to distinguish between "thread" and "thread ID".)

Reply:
I just tried to keep the structure of the ID correct -- to have exp_id.<string>. But yes, the naming still feels off. I'll try to fix it but I think we need to keep <string>.<string> as the invalid thread id as this is what the controller expects.

Reply:
Why? The controller should just expect an arbitrary thread_id and should not be trying to parse it ... right?

Reply:
Hmmm. True. I'll edit the controller appropriately. If thread id is not in the correct pattern, wouldn't it be preferable to throw an exception early (at the controller level itself). If this is the case, I'll add a check at the controller level, if not, should this check be at the service level?

Reply:
Done.

Reply:
Oh actually I realised why I had done exp_id.invalid.... The controller didn't parse the thread id but the decorator did. So I have to revert this to follow the thread_id structure. Any suggestions as to what I should do here?

Reply:
Decorator should validate the thread ID. If it can't parse it, it should error (probably 404 or 400). Otherwise it should parse it correctly and delegate accordingly.

There should be at least one test dedicated to testing unparseable thread IDs.